packages/common/rollup-plugins/writeBuildMetadata.js

```diff
@@ -0,0 +1,35 @@
+import { execSync } from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+
+/**
+ * Generates `build-meta.json` file in the given directory.
+ *
+ * @param {object} options
+ * @param {import("type-fest").PackageJson} options.pkg
+ * @param {string} options.outputDir
+ * @param {string} [options.cwd]
+ * @returns {import('rollup').Plugin}
+ */
+export const writeBuildMetadata = ({ pkg, outputDir, cwd = process.cwd() }) => ({
+  name: 'write-build-metadata',
+
+  writeBundle() {
+    const now = new Date();
+
+    fs.writeJSONSync(
+      path.resolve(cwd, outputDir, 'build-meta.json'),
+      {
+        name: pkg.name,
+        version: pkg.version,
+        buildDate: now.toLocaleString('en-US', { dateStyle: 'long' }),
+        buildTime: now.toLocaleString('en-US', { timeStyle: 'long' }),
+        gitCommit: execSync('git rev-parse HEAD').toString().trim(),
```

Comment:
👍  really useful info. I know we have "version" published in the plugin manifests, but I'd love to see something like this for plugins and apps (would help for debugging issues).